source/client/stream_decoder.cc

```diff
@@ -21,9 +21,10 @@ void StreamDecoder::decodeHeaders(Envoy::Http::ResponseHeaderMapPtr&& headers, b
   stream_info_.response_code_ = static_cast<uint32_t>(response_code);
   if (!latency_response_header_name_.empty()) {
     const auto timing_header_name = Envoy::Http::LowerCaseString(latency_response_header_name_);
-    const Envoy::Http::HeaderEntry* timing_header = response_headers_->get(timing_header_name);
-    if (timing_header != nullptr) {
-      absl::string_view timing_value = timing_header->value().getStringView();
+    const Envoy::Http::HeaderMap::GetResult& timing_header =
+        response_headers_->get(timing_header_name);
+    if (!timing_header.empty()) {
+      absl::string_view timing_value = timing_header[0]->value().getStringView();
```

Comment:
This could possibly use a comment or maybe even an assertion of sorts.  It looks like the envoy function was changed to now allow multiple HeaderEntries to be returned rather than one?

I gather that our use case would always expect only one - would it be reasonable to add a failure here if we received more than one?

Reply:
I added a sparse warning log line in dc0b618951bde50cc9eed7143fe39d0f7d433521 and also testing for that case to make sure we don't track timings when there's ambiguity because of multiple header/value pairs.